### ISSUE TITLE
Fluidsynth 2.4.1 problem with The Nervous Filter.mid

### ISSUE THREAD

**Issue by klerg** [FluidSynth/fluidsynth#1442]
### FluidSynth version
2.4.1

### Describe the bug
When the first 7 or so seconds of The Nervous Filter.mid is played in Fluidsynth 2.4.1 the audio drops out and sounds malformed and mangled

### Expected behavior
Fluidsynth should not dropout in the 1st 7 or so second when playing The Nervous Filter.mid  but sound like 2.4.0

### Steps to reproduce
Open Fluidsynth 2.4.1. load a soundfont, play The Nervous Filter.mid, and you can hear it will dropout for the 1st 7 or so seconds so it sounds malformed and mangled

### Additional context
I brought up this problem in #1430 before Fluidsynth 2.4.1 was released. I'm not sure was #1345 reverted in 2.4.1, or does this have to do with #1432, enabling NRPN 23 and 24, or something else, is any way to fix this issue ? Attached is The Nervous Filter.mid 

[TheNervousFilter.zip](https://github.com/user-attachments/files/17971005/TheNervousFilter.zip)

**Comment by derselbst** [comment]
There are no "dropouts" for me. This sounds like a problem with your soundcard. Pls. render it to an audio file instead:

```
fluidsynth -R 0 -C 0 -F "The Nervous Filter.ogg" "The Nervous Filter.mid" somesoundfont.sf2
```

If you hear dropouts, pls. upload the rendered file. If there are no dropouts, it's a problem on your system.

**Comment by klerg** [comment]
Ok so "dropouts" is not the right word. And it is more like clicks, cracks, and pops but it sounds better than 2.4.0, when I use this command it gives this error:

```
Rendering audio to file 'The Nervous Filter.ogg'..
fluidsynth: warning: Failed to determine audio file type from filename, defaulti
ng to WAV
```

But looks like it still works so I will upload it here, as the noise can clearly be heard in the 1st 7 seconds of The Nervous Filter.mid. I'm not sure if #1345 was reverted in 2.4.1 but overall it has less noise than 2.4.0. 

I will upload The Nervous Filter.mid played with two soundfonts, 1mgm.sf2 and gm.sf2, in 2.4.0 and 2.4.1. Any way to get rid of the _slow_ clicking and popping in the 1st 7 seconds of The Nervous Filter.mid in 2.4.1 ?

**Comment by derselbst** [comment]
I'll see what can be done, but it'll take time. Also to reproduce this, can you pls. provide `gm.sf2`? I know `1mgm.sf2`, I have it.

**Comment by derselbst** [comment]
Nevermind, I found it - it's GM_Roland.sf2

**Comment by klerg** [comment]
Sure that sounds good then, how long is that going to be ? Yes, gm.sf2 is the default gm.dls that comes with Windows converted to sf2 [here](
https://musical-artifacts.com/artifacts/713) is the link to get this file

Glad, you were able to find it. I noticed the 1st 7 seconds of The Nervous Filter.mid is less noisy with both 2GMGSMT.SF2 and 4GMGSMT.SF2 in 2.4.0 and 2.4.1. With 2.4.1 more noisy than 2.4.0 so here are the renderings, does this help at all ?

[The Nervous Filter2 2.4.0.zip](https://github.com/user-attachments/files/18049603/The.Nervous.Filter2.2.4.0.zip)

[The Nervous Filter2 2.4.1.zip](https://github.com/user-attachments/files/18049605/The.Nervous.Filter2.2.4.1.zip)

**Comment by derselbst** [comment]
IMO, the clicks have been fixed by #1444, which is now part of fluidsynth 2.4.2. Here are test renderings:

[The Nervous Filter.zip](https://github.com/user-attachments/files/18286237/The.Nervous.Filter.zip)

Test data can be found here: https://github.com/FluidSynth/testdata/tree/master/awe32_nrpn/Filter

This is the best what I can offer. Therefore, I'm closing this now.

**Comment by klerg** [comment]
Yes, I did not know that #1427 was the cause of this issue or that #1444 is the fix for it. I hear no more clicks, esp. w/ 1mgm.sf2

It looks like The Nervous Filter SB Live. wav is cut short I'll see if can upload the full audio.

Sure, it sounds much better than 2.4.1. It is a good time to close this case. I hope a fix for #1452 will improve it even more.

**Comment by ReinholdH** [comment]
Playback of "The Nervous Filter.mid" stops with reverb and chorus On for GeneralUser-GS.sf2 and fluidsynth 2.4.2 on Windows 7/10/11

On Windows 7/10/11 with the command 

fluidsynth -R 1 -C 1 "The Nervous Filter.mid" GeneralUser-GS.sf2
(GeneralUser-GS.sf2 from https://github.com/mrbumpy409/GeneralUser-GS) 

the playback completely stops with a loud crack after 10 - 20 seconds (the stopping time varies). The same stopping occurs with the recommended settings for the soundfont GeneralUser-GS.sf2 from @mrbumpy409 (see https://github.com/mrbumpy409/GeneralUser-GS/tree/main/documentation#302-fluidsynth). After such a stop a re-init of the fluidsynth is required. 

But with no reverb, no chorus

fluidsynth -R 0 -C 0 "The Nervous Filter.mid" GeneralUser-GS.sf2

there are some cracks but the playback completely finishes.

I observe the effect of the stopping only with GeneralUser-GS.sf2 but not with any other soundfont which I have tested. Another observation is that the issue is dependent on the used driver:
Interestingly, when exporting to a wav file with audio driver "file", all is fine. For "direct", "wasapi" (w/ and w/o exclusive mode) the stopping happens almost every time, for "jack" rarely and for "portaudio" w/ ASIO never.   
Tests have been made with various processors from very slow old PCs to very fast new ones. All PCs show the same effect.

Also see https://github.com/mrbumpy409/GeneralUser-GS/issues/8
